scripts/pkg/build_templates/opensearch/deb/debian/preinst

```diff
@@ -23,6 +25,29 @@ if command -v systemctl >/dev/null && systemctl is-active opensearch-performance
     systemctl --no-reload stop opensearch-performance-analyzer.service
 fi
 
+# Check if OPENSEARCH_INITIAL_ADMIN_PASSWORD is defined
+# TODO:
+# 1. This check will need to be modified if there will be a min dist for deb in future (currently there is none)
+# 2. Currently, the demo config setup is defined to run, in postinst, if `opensearch-security` is present. Cannot apply the same check here since the plugins folder is not available yet.
+
+# Check if this is an upgrade by checking whether opensearch already exists
+if dkpg-query -W opensearch >/dev/null 2>&1; then
```

Comment:
Per our offline discussion this has been changed from `-l` to `-W` for easier verification with return code, similar to `rpm -q`.

Reply:
it is `dpkg-query`, not `dkpg-query`.

Reply:
oops..fixed.